src/taco.ts

```diff
@@ -0,0 +1,82 @@
+import { DkgPublicKey, ThresholdMessageKit } from '@nucypher/nucypher-core';
+import { ethers } from 'ethers';
+
+import { ThresholdDecrypter } from './characters/cbd-recipient';
+import { Enrico } from './characters/enrico';
+import { Condition, ConditionExpression } from './conditions';
+import { DkgClient } from './dkg';
+import { getPorterUri } from './porter';
+import { toBytes } from './utils';
+
+export interface TacoMessageKit {
+  thresholdMessageKit: ThresholdMessageKit;
+  conditionExpr: ConditionExpression;
+  // TODO: How do we get rid of these two fields? We need them for decrypting
+  // We ritualId in order to fetch the DKG participants and create DecryptionRequests for them
+  ritualId: number;
+  // We need to know the threshold in order to create DecryptionRequests
+  threshold: number;
```

Comment:
Do we want to expose `class TacoMessageKit.requireSigner` so that decrypting users can detect when they need to pass signer in?

Reply:
Given my comment above, I question whether this `TacoMesageKit` object is needed at all.

conditionExpr is in the ACP in the ThresholdMessageKit, and I don't believe ritualId and threshold are needed because they can be obtained from the Coordinator contract along with the ThresholdMessageKit.

My concern lies in returning objects that aren't subject to versioning, and the caller potentially serializing them. Ideally, encrypt would return a `ThresholdMessageKit` just like python.

Of course, if additional information is actually needed in the `ThresholdMessageKit` itself we can explore that.